spinn_front_end_common/interface/interface_functions/chip_iobuf_extractor.py

```diff
@@ -46,9 +47,20 @@ def __init__(self, recovery_mode=False,
         self._recovery_mode = bool(recovery_mode)
 
     def __call__(
-            self, transceiver, executable_targets, executable_finder,
-            app_provenance_file_path, system_provenance_file_path,
-            binary_executable_types, from_cores="ALL", binary_types=None):
+            self, transceiver, executable_targets, executable_finder=None,
+            app_provenance_file_path=None, system_provenance_file_path=None,
+            binary_executable_types=None, from_cores="ALL", binary_types=None):
+
+        sim = get_simulator()
+        if executable_finder is None:
+            executable_finder = sim._executable_finder
+        if app_provenance_file_path is None:
+            app_provenance_file_path = sim._app_provenance_file_path
```

Comment:
This should become public if this is going to be used in this way, as otherwise someone might move it...